### PR TITLE
Wip/hughsie/plugins error

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/acpi-phat/fu-acpi-phat-plugin.c [hunk 1]
```diff
@@ -26,12 +26,23 @@ fu_acpi_phat_plugin_coldplug(FuPlugin *plugin, FuProgress *progress, GError **er
 	g_autofree gchar *str = NULL;
 	g_autoptr(FuFirmware) phat = fu_acpi_phat_new();
 	g_autoptr(GBytes) blob = NULL;
+	g_autoptr(GError) error_local = NULL;
 
 	path = fu_path_from_kind(FU_PATH_KIND_ACPI_TABLES);
 	fn = g_build_filename(path, "PHAT", NULL);
-	blob = fu_bytes_get_contents(fn, error);
-	if (blob == NULL)
+	blob = fu_bytes_get_contents(fn, &error_local);
```
Comment (superm1) [inline plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: Would it not be better to do this dance once in fu_bytes_get_contents?

Comment (hughsie) [reply plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: Yes and no! I've now move the fwupd_error_convert() _up_ there, but I think it makes sense to map the `FWUPD_ERROR_INVALID_FILE`->`FWUPD_ERROR_NOT_SUPPORTED` _down_ in the plugin still.

Comment (superm1) [reply plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: Why does that need to stay in the caller? What case does the caller not want that behavior?

Comment (hughsie) [reply plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: Some callers of fu_bytes_get_contents() explicitly want to handle the permission-denied or not-exists.

Comment (superm1) [reply plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: But couldn't they match the fwupd domain instead then?

Comment (hughsie) [reply plugins/acpi-phat/fu-acpi-phat-plugin.c:33]: Right, that's what I did in libfwupdplugin/fu-config.c in this branch too -- but not-supported is a conceptual meaning from the plugin -- if fu_bytes_get_contents() returned not-supported we don't know if the read action was supported, or if the file was just missing.

**Review by superm1 — APPROVED** [review]

**Pull request merged**